src/views/bookmark_x/controller.ts

```diff
@@ -118,12 +149,42 @@ export class Controller {
         if (SpaceSortItem.sorting_item !== undefined) {
             let old_tvitem = SpaceSortItem.sorting_item;
             old_tvitem.contextValue = SpaceSortItem.sorting_ctx;
+            if (this.using_active_icon(old_tvitem)) {
+                // the deselected item is using active group icon now.
+                old_tvitem.iconPath = ICON_ACTIVE_GROUP;
+            } else {
+                switch (old_tvitem.base!.type) {
+                    case ITEM_TYPE_GROUP:
+                    case ITEM_TYPE_GROUPBM:
+                        old_tvitem.iconPath = ICON_GROUP;
+                        break;
+                    case ITEM_TYPE_BM:
+                        old_tvitem.iconPath = ICON_BOOKMARK;
+                        break;
+                    default:
+                        break;
+                }
+            }
```

Comment:
add a class to do the dirty thing.
In principle, if this function can be done outside the Controller, don't do it in the Controller.